substrate/primitives/trie/src/cache/shared_cache.rs

```diff
@@ -602,19 +652,29 @@ impl<H: Hasher> SharedTrieCacheInner<H> {
 /// bounds given via the [`CacheSize`] at startup.
 ///
 /// The instance of this object can be shared between multiple threads.
-pub struct SharedTrieCache<H: Hasher> {
+pub struct SharedTrieCache<H: Hasher + 'static> {
 	inner: Arc<RwLock<SharedTrieCacheInner<H>>>,
+	/// The channel used by the writeback thread to wait for the trigger that it needs to start
+	/// promoting the local trie cache accesses to the shared trie cache. This is used only when
+	/// `promote_in_background` is passed to the constructor.
+	writeback_thread_rx: Arc<Mutex<Option<std::sync::mpsc::Receiver<bool>>>>,
+	/// The channel used to trigger the writeback thread.
+	tx: Option<std::sync::mpsc::Sender<bool>>,
 }
 
 impl<H: Hasher> Clone for SharedTrieCache<H> {
 	fn clone(&self) -> Self {
-		Self { inner: self.inner.clone() }
+		Self {
+			inner: self.inner.clone(),
+			tx: self.tx.clone(),
+			writeback_thread_rx: self.writeback_thread_rx.clone(),
+		}
 	}
 }
 
-impl<H: Hasher> SharedTrieCache<H> {
+impl<H: Hasher + 'static> SharedTrieCache<H> {
 	/// Create a new [`SharedTrieCache`].
-	pub fn new(cache_size: CacheSize) -> Self {
+	pub fn new(cache_size: CacheSize, promote_in_background: bool) -> Self {
```

Comment:
According to the changes promote_in_background == unlimited_local_cache, but it's not obvious here.